Fix/disable dry-run of the publish action on `next`

On `next`, the `publish-crates-dry-run` job introduced in https://github.com/0xMiden/miden-client/pull/1403 is [failing every time](https://github.com/0xMiden/miden-client/actions/runs/18942594344), particularly because we are using non-published crates (eg from `miden-base`):

```
Run cargo publish --workspace --exclude miden-client-web --dry-run
info: syncing channel updates for '1.90-x86_64-unknown-linux-gnu'
info: latest update on 2025-09-18, rust version 1.90.0 (1159e78c4 2025-09-14)
info: downloading component 'cargo'
info: downloading component 'clippy'
info: downloading component 'rust-src'
info: downloading component 'rust-std' for 'wasm32-unknown-unknown'
info: downloading component 'rust-std'
info: downloading component 'rustc'
info: downloading component 'rustfmt'
info: installing component 'cargo'
info: installing component 'clippy'
info: installing component 'rust-src'
info: installing component 'rust-std' for 'wasm32-unknown-unknown'
info: installing component 'rust-std'
info: installing component 'rustc'
info: installing component 'rustfmt'
    Updating crates.io index
error: failed to verify manifest at `/home/runner/work/miden-client/miden-client/crates/rust-client/Cargo.toml`

Caused by:
  all dependencies must have a version requirement specified when publishing.
  dependency `miden-lib` does not specify a version
  Note: The published dependency will use the version from crates.io,
  the `git` specification will be removed from the dependency declaration.
Error: Process completed with exit code 101.
```

We probably want to disable this action on `next` due to this. 

Additionally, we should also probably add the workflow to run on PRs to `main` and not on merges

--- a/.github/workflows/publish-crates-dry-run.yml
+++ b/.github/workflows/publish-crates-dry-run.yml
@@ -2,7 +2,7 @@ name: Dry Run Publish Rust Crates
 
 on:
   push:
-    branches: [main, next]
+    branches: [main]
 
 permissions:
   contents: read

